### PR TITLE
API/UI split

### PR DESCRIPTION
**Pull request by Vanders**
Refactor everything so that the Server, UI & Sidekiq are all built from seperate Dockerfiles.
Make the UI work reliably by setting SESSION_SECRET in the environment.

### CODE REVIEW

File: cyclid_sidekiq/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM ubuntu:trusty
+
+RUN apt-get update
+RUN apt-get install -qy software-properties-common
+RUN add-apt-repository ppa:brightbox/ruby-ng
+RUN apt-get update
+RUN apt-get install -qy ruby2.3 ruby2.3-dev build-essential cmake mysql-client libmysqlclient-dev libsasl2-dev
+RUN mkdir -p /var/lib/cyclid /var/run/cyclid /etc/cyclid /var/log/cyclid
+COPY files/cyclid-0.2.4.gem /tmp/
```
Comment (Afterglow) [inline cyclid_sidekiq/Dockerfile]: any chance you can release this to rubygems?

File: cyclid_ui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM ubuntu:trusty
+
+RUN apt-get update
+RUN apt-get install -qy software-properties-common
+RUN add-apt-repository ppa:brightbox/ruby-ng
+RUN apt-get update
+RUN apt-get install -qy ruby2.3 ruby2.3-dev build-essential cmake mysql-client libmysqlclient-dev libsasl2-dev
+RUN mkdir -p /var/run/cyclid /etc/cyclid /var/log/cyclid /var/lib/cyclid-ui
+COPY files/cyclid-ui-0.2.2.gem /tmp/
```
Comment (Afterglow) [inline cyclid_ui/Dockerfile]: release this too?

File: Dockerfile [hunk 3]
```diff
@@ -4,7 +4,7 @@ RUN apt-get update
 RUN apt-get install -qy software-properties-common
 RUN add-apt-repository ppa:brightbox/ruby-ng
 RUN apt-get update
-RUN apt-get install -qy ruby2.3 ruby2.3-dev build-essential cmake mysql-client libmysqlclient-dev libsasl2-dev
+RUN apt-get install -qy ruby2.3 ruby2.3-dev build-essential cmake mysql-client libmysqlclient-dev libsasl2-dev nginx
```
Comment (Afterglow) [inline Dockerfile]: is nginx necessary, i'm not sure it actually gets run at any point?

File: cyclid_ui/files/unicorn.rb [hunk 4]
```diff
@@ -5,6 +5,7 @@
 stdout_path "/var/log/cyclid/unicorn.cyclid-ui.log"
 
 listen 80
+#listen "/var/run/unicorn.cyclid-ui.sock"
```
Comment (Afterglow) [inline cyclid_ui/files/unicorn.rb]: remove this?

**Review by Afterglow — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (Vanders) [reply Dockerfile]: Yes nginx can be dropped.

File: cyclid_ui/files/unicorn.rb (hunk 4 above)
Comment (Vanders) [reply cyclid_ui/files/unicorn.rb]: Yup, missed that one.

File: cyclid_ui/Dockerfile (hunk 2 above)
Comment (Vanders) [reply cyclid_ui/Dockerfile]: Those are both custom Gems; I want to roll up some other changes and do a patch release of both Cyclid & Cyclid UI; maybe next week? I can open a PR to install from Rubygems once that's done.

**Review by Afterglow — APPROVED** [review]

**Pull request merged**